pandas/core/generic.py

```diff
@@ -6827,6 +6827,10 @@ def fillna(
         reindex : Conform object to new index.
         asfreq : Convert TimeSeries to specified frequency.
 
+        Notes
+        -----
+        For non-object dtype, ``value=None`` will use the NA value of the dtype.
+
         Examples
```

Comment:
I think this is a good suggestion, but what do you think about doing it in https://pandas.pydata.org/docs/user_guide/missing_data.html#filling-missing-data instead of the docstring here?

Reply:
> I think this is a good suggestion, but what do you think about doing it in https://pandas.pydata.org/docs/user_guide/missing_data.html#filling-missing-data instead of the docstring here? Can then link to this in the docstring.

That's fine, although I think the behavior of `None` with respect to how it is treated as a missing value isn't necessarily well explained on that page, so maybe that should be done as well.

Reply:
Added docs:

![image](https://github.com/pandas-dev/pandas/assets/45562402/2ec84d5c-84ef-47e1-8383-5391c0e67f2e)

![image](https://github.com/pandas-dev/pandas/assets/45562402/af90ec8e-fedb-4566-b188-9d774cf86dad)